src/ripple/app/misc/SHAMapStoreImp.cpp

```diff
@@ -653,9 +652,16 @@ SHAMapStoreImp::freshenCaches()
         return;
 }
 
-bool
+void
 SHAMapStoreImp::clearPrior(LedgerIndex lastRotated)
 {
+    if (app_.config().reporting())
+    {
+        assert(false);
+        Throw<std::runtime_error>(
+            "Reporting does not support online_delete. Remove "
+            "online_delete info from config");
+    }
```

Comment:
There's presently only one call to this method, from `run`, which already has this guard at its entry. Are we copying it here to protect against someone adding another call in the future? I don't mind, but I don't want you to feel like you have to put it here; I think the chance of it ever being hit is virtually nil.

Reply:
Yeah I am just being extra cautious here, in case the code is changed in the future. I know it's probably not necessary in this spot, but I sprinkled this check in various places, some of which are redundant, just to be extra careful.